managers.py

```diff
@@ -613,6 +556,7 @@ def handle_data(self, this_start, this_end, spectra):
                 self.data_handler.main(
                     self.datalog, this_start, this_end, average_data=average_data)
             else:
+                self.data_log(self.datalog, average_data=average_data)
                 return average_data
 
```

Comment:
I think you could potentially clean up even more by initializing average_data outside the if statements for sensor_type and then running the data_handler.main/data_log if/else clause outside those if statements (and therefore only having to include that text once).

Reply:
You're right I was able to do that! I also changed one other thing in that handle_data function so that its a little cleaner again.